include/gfx/convert.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of RGBDS.
+ *
+ * Copyright (c) 2022, Eldred Habert and RGBDS contributors.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#ifndef RGBDS_GFX_CONVERT_HPP
+#define RGBDS_GFX_CONVERT_HPP
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "gfx/main.hpp"
+
+struct Rgba {
+	uint8_t red;
+	uint8_t green;
+	uint8_t blue;
+	uint8_t alpha;
+
+	Rgba(uint8_t r, uint8_t g, uint8_t b, uint8_t a) : red(r), green(g), blue(b), alpha(a) {}
+	Rgba(uint32_t rgba) : red(rgba), green(rgba >> 8), blue(rgba >> 16), alpha(rgba >> 24) {}
+
+	operator uint32_t() const {
+		return toCSS();
+	}
+	/**
+	 * Returns this RGBA as a 32-bit number that can be printed in hex (`%08x`) to yield its CSS
+	 * representation
+	 */
+	uint32_t toCSS() const {
```

Comment:
"toCSS" is a misnomer; it's "to number which could be formatted in many ways, including `#%08x` like CSS", but you may as well use `#%02x%02x%02x%02x` with the `red/green/blue/alpha` components for that. I think the meaning of converting it to a `uint32_t` is clear on its own.
```suggestion
```

Reply:
No, because the components were previously ARGB, which does not print in the CSS format. That's when I introduced this name.

Reply:
I mean, if `(uint32_t)` casting is going to behave the same way as `toCSS`, I don't see a need for both. Separate `toCSS` would make sense if `(uint32_t)` still returned ARGB order. The `operator uint32_t` could still have a comment that its RGBA order is chosen to easily print CSS-style RGBA colors as `#%08x`.

Reply:
I think the conversion operator is only used for the comparison below; if no other remnants remain, then we can remove it and use `.toCSS()` in the comparison operator.